File: baekjoon/14499.py
def move(dr):
    if dr == 1:
        if now[1] == m-1:
            return -1
        now[1] += 1
        now[2], now[3], now[4] = 7 - now[3], now[2], now[4]
    elif dr == 2:
        if now[1] == 0:
            return -1
        now[1] -= 1
        now[2], now[3], now[4] = now[3], 7-now[2], now[4]
    elif dr == 3:
        if now[0] == 0:
            return -1
        now[0] -= 1
        now[2], now[3], now[4] = now[4], now[3], 7-now[2]
    else:
        if now[0] == n-1:
            return -1
        now[0] += 1
        now[2], now[3], now[4] = 7-now[4], now[3], now[2]
    if lst[now[0]][now[1]]:
        num[7-now[2]] = lst[now[0]][now[1]]
        lst[now[0]][now[1]] = 0
    else:
        lst[now[0]][now[1]] = num[7-now[2]]
    return num[now[2]]


n, m, x, y, k = map(int, input().split())
lst = list(list(map(int, input().split())) for _ in range(n))
task = list(map(int, input().split()))
now = [x, y, 1, 3, 5]
num = [0, 0, 0, 0, 0, 0, 0]
for i in task:
    d = move(i)
    if d >= 0:
        print(d)

File: c/Project1/Project1/14499.cpp
/*
#include <iostream>
#include <vector>
using namespace std;
int n, m, x, y, nx, ny, k, d, now[3] = { 1, 3, 5 }, num[7]{}, dr[4][2] = { {1, 0}, {0, 1}, {0, -1}, {-1, 0} };
vector<int> lst[20];
void move(int);

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cin >> n >> m >> x >> y >> k;
	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < m; j++)
		{
			cin >> d;
			lst[i].push_back(d);
		}
	}
	for (int i = 0; i < k; i++)
	{
		cin >> d;
		move(d);
	}
	return 0;
}

void move(int d) {
	nx = x + dr[d % 4][0];
	ny = y + dr[d % 4][1];
	if (0 > nx || nx >= n || ny < 0 || ny >= m) {
		return;
	}
	if (d == 1)
	{
		int d1 = 7 - now[1], d2 = now[0], d3 = now[2];
		now[0] = d1;
		now[1] = d2;
		now[2] = d3;
	}
	else if (d == 2)
	{
		int d1 = now[1], d2 = 7 - now[0], d3 = now[2];
		now[0] = d1;
		now[1] = d2;
		now[2] = d3;
	}
	else if (d == 3)
	{
		int d1 = now[2], d2 = now[1], d3 = 7 - now[0];
		now[0] = d1;
		now[1] = d2;
		now[2] = d3;
	}
	else
	{
		int d1 = 7 - now[2], d2 = now[1], d3 = now[0];
		now[0] = d1;
		now[1] = d2;
		now[2] = d3;
	}
	x = nx;
	y = ny;
	if (lst[x][y] == 0)
	{
		lst[x][y] = num[7 - now[0]];
	}
	else
	{
		num[7 - now[0]] = lst[x][y];
		lst[x][y] = 0;
	}
	cout << num[now[0]] << endl;
}
*/